### PR TITLE
adding fgap recipe

### PR DESCRIPTION
**Pull request by nickp60**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).
Hi all, this represents the first (to my knowledge) use of an octave script in bioconda.  What I have done is use the build stage to create a runner script that tells octave where to find the source and passes the appropriate args. Comments and critiques welcome;  I am aware this is far from optimal, but lets consider it a starting point!

https://github.com/pirovc/fgap

### CODE REVIEW

File: recipes/fgap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+build:
+  number: 0
+  skip: True  # [osx]
+  
+package:
+  name: fgap
+  version: 1.8.1
+source:
+  fn: fgap-1.8.1.tar.gz
+  # this is nickp60's fork because bioconda needs tagged releases
```
Comment (mbargull) [inline recipes/fgap/meta.yaml]: It's totally fine to use Git archives for certain commits, e.g., for this project that would be:
```yaml
  url: https://github.com/pirovc/fgap/archive/e3004998f9058e8505b7812544d4c1dbd595a169.tar.gz
```
That way you won't have to maintain a fork only to have tags.
Additionally, there is also nothing wrong by asking upstream if they could tag their releases in the future :wink:

Comment (nickp60) [reply recipes/fgap/meta.yaml]: @mbargull ooo, I did not know about Git archives!  I move to that until upstream makes a release.  thanks for the tip!

File: recipes/fgap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  summary: "FGAP: an automated gap closing tool"
+build:
+  number: 0
+  skip: True  # [osx]
+  
+package:
+  name: fgap
+  version: 1.8.1
+source:
+  fn: e3004998f9058e8505b7812544d4c1dbd595a169.tar.gz
```
Comment (mbargull) [inline recipes/fgap/meta.yaml]: You can still use 
```yaml
  fn: fgap-1.8.1.tar.gz
```
`conda-build` would default to `e3004998f9058e8505b7812544d4c1dbd595a169.tar.gz` if `fn` isn't given, but we can also use a more meaningful name like `fgap-1.8.1.tar.gz` if we want to.

File: recipes/fgap/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+echo "#!/bin/bash" >> FGAP
+ls
+echo "echo 'starting FGAP'" >> FGAP
```
Comment (mbargull) [inline recipes/fgap/build.sh]: Do we really need the `echo 'starting FGAP'` here?
Couldn't we just use something like:
```bash
cat > FGAP <<EOF
#!/bin/bash
octave --path "$PREFIX/bin/" --no-gui --eval "fgap \$@"
EOF
```
?

Comment (mbargull) [reply recipes/fgap/build.sh]: [Sorry that I added those comments after each other and not as part of a full review. Clicked the wrong button..]

**Comment by mbargull** [comment]
Nice! Thank you, @nickp60!

**Pull request merged**